File: File Uploads/Multipart Example.m


NSString *deviceID = [[NSUserDefaults standardUserDefaults] objectForKey:@"deviceID"];
NSString *textContent = @"This is a new note";

// Build the request body
NSString *boundary = @"SportuondoFormBoundary";
NSMutableData *body = [NSMutableData data];
// Body part for "deviceId" parameter. This is a string.
[body appendData:[[NSString stringWithFormat:@"--%@\r\n", boundary] dataUsingEncoding:NSUTF8StringEncoding]];
[body appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=\"%@\"\r\n\r\n", @"deviceId"] dataUsingEncoding:NSUTF8StringEncoding]];
[body appendData:[[NSString stringWithFormat:@"%@\r\n", deviceID] dataUsingEncoding:NSUTF8StringEncoding]];
// Body part for "textContent" parameter. This is a string.
[body appendData:[[NSString stringWithFormat:@"--%@\r\n", boundary] dataUsingEncoding:NSUTF8StringEncoding]];
[body appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=\"%@\"\r\n\r\n", @"textContent"] dataUsingEncoding:NSUTF8StringEncoding]];
[body appendData:[[NSString stringWithFormat:@"%@\r\n", textContent] dataUsingEncoding:NSUTF8StringEncoding]];
// Body part for the attachament. This is an image.
NSData *imageData = UIImageJPEGRepresentation([UIImage imageNamed:@"ranking"], 0.6);
if (imageData) {
    [body appendData:[[NSString stringWithFormat:@"--%@\r\n", boundary] dataUsingEncoding:NSUTF8StringEncoding]];
    [body appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=\"%@\"; filename=\"image.jpg\"\r\n", @"image"] dataUsingEncoding:NSUTF8StringEncoding]];
    [body appendData:[@"Content-Type: image/jpeg\r\n\r\n" dataUsingEncoding:NSUTF8StringEncoding]];
    [body appendData:imageData];
    [body appendData:[[NSString stringWithFormat:@"\r\n"] dataUsingEncoding:NSUTF8StringEncoding]];
}
[body appendData:[[NSString stringWithFormat:@"--%@--\r\n", boundary] dataUsingEncoding:NSUTF8StringEncoding]];

// Setup the session
NSURLSessionConfiguration *sessionConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
sessionConfiguration.HTTPAdditionalHeaders = @{
                                               @"api-key"       : @"55e76dc4bbae25b066cb",
                                               @"Accept"        : @"application/json",
                                               @"Content-Type"  : [NSString stringWithFormat:@"multipart/form-data; boundary=%@", boundary]
                                               };

// Create the session
// We can use the delegate to track upload progress
NSURLSession *session = [NSURLSession sessionWithConfiguration:sessionConfiguration delegate:self delegateQueue:nil];

// Data uploading task. We could use NSURLSessionUploadTask instead of NSURLSessionDataTask if we needed to support uploads in the background
NSURL *url = [NSURL URLWithString:@"URL_TO_UPLOAD_TO"];
NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
request.HTTPMethod = @"POST";
request.HTTPBody = body;
NSURLSessionDataTask *uploadTask = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
    // Process the response
}];
[uploadTask resume];

File: File Uploads/Multipart Example.swift


let requestURL = NSURL(string:"http://leonbaird.co.uk/iphone/process.php")!
let imageData = UIImageJPEGRepresentation(imagePreview.image, 1.0)!
let descriptionText = descriptionTextField.text // get from outlet to UITextField

let boundary = "----LeonBairdUploader1234567"
let seperator = "\r\n--\(boundary)\r\n".dataUsingEncoding(
    NSUTF8StringEncoding, allowLossyConversion:false)!

// make the body
var bodyData = NSMutableData()
bodyData.appendData(seperator)
bodyData.appendData("Content-Disposition: form-data; name=\"description\"".dataUsingEncoding(
    NSUTF8StringEncoding, allowLossyConversion:false)!)
bodyData.appendData(descriptionText.dataUsingEncoding(
    NSUTF8StringEncoding, allowLossyConversion:false)!)
bodyData.appendData(seperator)
bodyData.appendData("Content-Disposition: form-data; name=\"image[]\"; filename=\"uploaded_image.jpg\"\r\n".dataUsingEncoding(
    NSUTF8StringEncoding, allowLossyConversion:false)!)
bodyData.appendData("Content-Type: application/octet-stream\r\n\r\n".dataUsingEncoding(
    NSUTF8StringEncoding, allowLossyConversion:false)!)
bodyData.appendData(imageData)
bodyData.appendData("\r\n--\(boundary)--\r\n".dataUsingEncoding(
    NSUTF8StringEncoding, allowLossyConversion:false)!)